pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -458,3 +470,52 @@ func (vc *VirtualCenter) GetDatastoresByCluster(ctx context.Context, clusterMore
 	}
 	return dsList, nil
 }
+
+// GetVirtualCenterInstance returns the vcenter object singleton.
+// It is thread safe.
+// Takes in a boolean paramater reloadConfig.
+// If reloadConfig is true, the vcenter object is instantiated again and the old object becomes eligible for garbage collection.
+// If reloadConfig is false and instance was already initialized, the previous instance is returned.
+func GetVirtualCenterInstance(ctx context.Context, config *config.ConfigurationInfo, reloadConfig bool) (*VirtualCenter, error) {
```

Comment:
`reloadConfig` ==> How about renaming to `reinitialize` ?

Reply:
done changed signature to

```
func GetVirtualCenterInstance(ctx context.Context, config *config.ConfigurationInfo, reinitialize bool) (*VirtualCenter, error) {
```